pkg/reader/usdc_reader.go

```diff
@@ -269,14 +288,14 @@ func NewSourceTokenDataPayload(nonce uint64, sourceDomain uint32) *SourceTokenDa
 }
 
 func NewSourceTokenDataPayloadFromBytes(extraData cciptypes.Bytes) (*SourceTokenDataPayload, error) {
-	if len(extraData) < 12 {
-		return nil, fmt.Errorf("extraData is too short, expected at least 12 bytes")
+	if len(extraData) < 64 {
+		return nil, fmt.Errorf("extraData is too short, expected at least 64 bytes")
 	}
 
-	// Extract the nonce (first 8 bytes)
-	nonce := binary.BigEndian.Uint64(extraData[:8])
-	// Extract the sourceDomain (next 4 bytes)
-	sourceDomain := binary.BigEndian.Uint32(extraData[8:12])
+	// Extract the nonce (first 8 bytes), padded to 32 bytes
```

Comment:
nit might be worth mentioning that this is due to EVM abi encoding

Reply:
Or maybe the whole func is evm specific

Reply:
I will clarify that